geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementService.java

```diff
@@ -35,7 +35,8 @@
    *        cluster, as well as the group this config belongs to
    * @see CacheElement
    */
-  ClusterManagementResult create(CacheElement config);
+  <T extends RestfulEndpoint & RespondsWith<R>, R extends CMSJsonSerializable> ClusterManagementResult<T> create(
```

Comment:
I really don't see the need for this level of separation at this point.  create/delete/update/list are designed for operations on CacheElement which is by nature Groupable and JsonSerializable. I would say let's hold off this separation until we see a real need for it to avoid unnecessary cognitive burden of the code.